lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,351 @@
+
+There are also a list of commands and phases that you can call. To see the full
+list you can run:
+    
+.. code-block:: console
+    
+    $ python setup.py --help-commands
+
+For each of these commands, Spack has a method that can be overridden. 
+
+Using the Package class is possible but it brings with it a lot of unnecessary
+methods that are useless for a Python package. Instead, Spack has a PythonPackage
+subclass that allows packagers of Python modules to be able to invoke setup.py
+and use Distutils.  
+
+We will write a package file for Pandas_:
+
+.. _pandas: https://pandas.pydata.org
+
+.. code-block:: console
+    
+    $ spack create -f https://pypi.io/packages/source/p/pandas/pandas-0.19.0.tar.gz
+    ==> This looks like a URL for pandas
+    ==> Warning: Spack was unable to fetch url list due to a certificate verification problem. You can try running spack -k, which will not check SSL certificates. Use this at your own risk.
+    ==> Found 1 version of pandas:
+
+    0.19.0  https://pypi.io/packages/source/p/pandas/pandas-0.19.0.tar.gz
+
+    ==> How many would you like to checksum? (default is 1, q to abort) 1
+    ==> Downloading...
+    ==> Fetching https://pypi.io/packages/source/p/pandas/pandas-0.19.0.tar.gz
+    ######################################################################## 100.0%
+    ==> Checksummed 1 version of pandas
+    ==> This package looks like it uses the python build system
+    ==> Changing package name from pandas to py-pandas
+    ==> Created template for py-pandas package
+    ==> Created package file: /Users/mamelara/spack/var/spack/repos/builtin/packages/py-pandas/package.py
+
+And we are left with the following template:
+
+.. literalinclude:: tutorial/examples/PyPackage/0.package.py
+
+As you can see this is not any different than any package template that we have
+written. Spack will provide sensible defaults and we have the option of providing
+it.
+
+Luckily for us, there is no need to provide build args. The default args
+that Spack provides is 'build' which builds everything needed to install. Once
+the build phase is completed the install phase will commence and soon we will
+have our python module. 
+
+Here is the completed Pandas script:
+
+.. literalinclude:: tutorial/examples/PyPackage/1.package.py
+
+As you can see, Pandas is easy to install since we do not have to do anything
+rather convoluted and complex. Instead, Spack has provided us the right abstractions
+to make installing Python packages easy.
+
+-------------------
+Other Build Systems
+-------------------
+
+Although we won't get in depth with any of the other build systems that Spack
+has support for me it is worth mentioning that Spack does provide subclasses
+for the following build systems:
+
+IntelPackage, SconsPackage, WafPackage, RPackage, PerlPackage, and QMake.
+
+
+Each of these classes have their own abstractions to help assist the packager
+write for these types of packages. 
+
```

Comment:
Maybe note here that the generic Package class is available for anything that doesn't fit nicely into one of the other classes?